fold-this.el

```diff
@@ -25,36 +26,53 @@
 ;;
 ;; ## How it works
 ;;
-;; The command `fold-this` visually replaces the current region with `...`.
+;; The command `fold-this` visually replaces the current region with `[[…]]`.
 ;; If you move point into the ellipsis and press enter or `C-g` it is unfolded.
 ;;
 ;; You can unfold everything with `fold-this-unfold-all`.
 ;;
 ;; You can fold all instances of the text in the region with `fold-this-all`.
 ;;
-;; ## Setup
-;;
-;; I won't presume to know which keys you want these functions bound to,
-;; so you'll have to set that up for yourself. Here's some example code,
-;; which incidentally is what I use:
-;;
-;;     (global-set-key (kbd "C-c C-f") 'fold-this-all)
-;;     (global-set-key (kbd "C-c C-F") 'fold-this)
-;;     (global-set-key (kbd "C-c M-f") 'fold-this-unfold-all)
-
 ;;; Code:
-
 (require 'thingatpt)
 
-(defvar fold-this-keymap (make-sparse-keymap))
-(define-key fold-this-keymap (kbd "<return>") 'fold-this-unfold-at-point)
-(define-key fold-this-keymap (kbd "C-g") 'fold-this-unfold-at-point)
+(defgroup fold-this nil
+  "Just fold this region please."
+  :prefix "fold-this-"
+  :group 'languages)
+
+(defcustom fold-this-mode-key-prefix (kbd "C-c")
+  "The prefix key for `fold-this-mode' commands."
+  :group 'fold-this
+  :type 'sexp)
+
+(defvar fold-this--overlay-keymap
+  (let ((map (make-sparse-keymap)))
+    (define-key map (kbd "<return>") 'fold-this-unfold-at-point)
+    (define-key map (kbd "C-g") 'fold-this-unfold-at-point)
+    map)
+  "Keymap for `fold-this' but only on the overlay.")
+
+(defvar fold-this-keymap
+  (let ((map (make-sparse-keymap)))
+    (let ((prefix-map (make-sparse-keymap)))
+      (define-key prefix-map (kbd "C-f") 'fold-this)
+      (define-key prefix-map (kbd "M-C-f") 'fold-this-all)
+      (define-key prefix-map (kbd "M-f") 'fold-this-unfold-all)
+    (define-key map fold-this-mode-key-prefix prefix-map)
+    map))
+  "Keymap for `fold-this'.")
 
 (defface fold-this-overlay
-  '((t (:inherit default)))
+  '((t (:inherit default :foreground "green")))
   "Face used to highlight the fold overlay."
   :group 'fold-this)
 
+(defcustom fold-this-overlay-text "[[…]]"
```

Comment:
I'm a bit worried about the unicode here, some terminals might still have problem with this.

Reply:
It's a mistake. I was thinking of the plain underscore.
Sorry about that.
I've also some others patches to avoid dupes on the overlay alist. I hope to send you another PR on the WE.